pkg/test/framework/features/README.md

```diff
@@ -0,0 +1,45 @@
+# Overview
+The features package defines values that are used to track feature coverage at eng.istio.io
+# Labeling a Test
+Adding a label to a test using the integration framework is easy.  Simply call the Feature() function on the Test Object, passing any relevant feature constants as parameters.  If you are testing a feature that does not have a constant defined, see the next section for how to define one.
+# Adding New Feature Constants
+For consistency, feature constants are generated from `features.yaml`.  Each entry in this file will be generated into a dot delimited feature label.  For instance:
+
+    foo:
+      bar
+      baz:
+        boo
+will generate constants like
+
+    const (
+        foo_bar = "foo.bar"
```

Comment:
Why? I know we do this in other parts of the code, but I don't get the need to write yaml -> go codegen. Its just as simple to have it inline in the go code in my opinion, and then we don't need yet another codegen

Reply:
If it were inline in go, there would be lots of repetition.  In yaml, adding a feature at 'Observability.Telemetry.Grafana.DefaultIntegration' involves only adding the 'DefaultIntegration' string (assuming the rest of the hierarchy already exists.  In go, the entire string would need to be typed, introducing a near certainty over time of typos causing report divergence.

Reply:
@howardjohn 

I have no objection to authoring directly in Go if we can achieve a reasonably readable and concise structure. The closest I could see was anonymous structs.

Thoughts?